packages/context/src/resolver.ts

```diff
@@ -23,10 +23,34 @@ export type Constructor<T> = new(...args: any[]) => T;
  */
 export function instantiateClass<T>(ctor: Constructor<T>, ctx: Context): T | Promise<T> {
   const argsOrPromise = resolveInjectedArguments(ctor, ctx);
+  const propertiesOrPromise = resolveInjectedProperties(ctor, ctx);
+  let inst: T | Promise<T>;
   if (isPromise(argsOrPromise)) {
-    return argsOrPromise.then(args => new ctor(...args));
+    // Instantiate the class asynchronously
+    inst = argsOrPromise.then(args => new ctor(...args));
   } else {
-    return new ctor(...argsOrPromise);
+    // Instantiate the class synchronously
+    inst = new ctor(...argsOrPromise);
+  }
+  if (isPromise(propertiesOrPromise)) {
+    return propertiesOrPromise.then((props) => {
+      if (isPromise(inst)) {
+        // Inject the properties asynchrounously
+        return inst.then((obj) => { Object.assign(obj, props); return obj; });
+      } else {
+        // Inject the properties synchrounously
+        Object.assign(inst, props);
+        return inst;
+      }
+    });
+  } else {
+    if (isPromise(inst)) {
+      // Inject the properties asynchrounously
+      return inst.then(obj => Object.assign(obj, propertiesOrPromise));
```

Comment:
This is inconsistent with L39. Why is `return obj` needed on L39 but not needed here?

Similarly for L43, why is `return inst` needed?